repository_service_tuf/cli/artifact/repository.py

```diff
@@ -0,0 +1,278 @@
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+def add(
+    context: Context,
+    name: str,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+) -> None:
+    """
+    Add a new repository.
+    """
+
+    rstuf_config = context.obj.get("settings").as_dict()
+
+    success_msg: str = f"Successfully added {name} repository to config"
+    if rstuf_config.get("REPOSITORIES") and rstuf_config["REPOSITORIES"].get(
+        name
+    ):
+        console.print(f"Repository {name} already configured")
+        success_msg = f"Successfully updated repository {name}"
+
+    encoded_root = b""
+    if root:
+        encoded_root = base64.b64encode(bytes(root, "utf-8"))
+
+    repo_data = {
+        "artifact_base_url": artifacts_url,
+        "metadata_url": metadata_url,
+        "trusted_root": encoded_root,
+        "hash_prefix": hash_prefix,
+    }
+
+    if rstuf_config.get("REPOSITORIES"):
+        rstuf_config["REPOSITORIES"][name] = repo_data
+    else:
+        rstuf_config["REPOSITORIES"] = {name: repo_data}
+
+    if context.obj.get("config"):
+        write_config(context.obj.get("config"), rstuf_config, False)
+        console.print(success_msg)
+
+
+@repository.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+@click.argument("repository")
+def update(
+    context: Context,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+    repository: str,
+) -> None:
+    """
+    Update repository.
+    """
+
+    rstuf_config = context.obj.get("settings").as_dict()
+
+    if not rstuf_config.get("REPOSITORIES"):
+        raise click.ClickException(
+            "There are no configured repositories to update"
+        )
+
+    if not rstuf_config["REPOSITORIES"].get(repository):
+        raise click.ClickException(
+            f"Repository {repository} not available in config. "
+            "You can create it instead"
+        )
+
+    if root:
+        rstuf_config["REPOSITORIES"][repository][  # pragma: no cover
```

Comment:
Maybe you don't need `pragma`

Reply:
I have test coverage for this use-case, but codecov doesn't find it properly.

Reply:
I couldn't find a test where you test that.
I think you should add a test for that.